tests/authentication_tests/test_ulf2.py

```diff
@@ -363,9 +367,31 @@ def api_handler1(request):
             "Unexpected response from the ShotGrid site",
         )
 
+        # Send a 200 with sessionRequestId but not url field
+        self.httpd.router["[POST]/internal_api/app_session_request"] = lambda request: {
+            "json": {
+                "sessionRequestId": "a1b2c3",
+            }
+        }
+
+        with self.assertRaises(unified_login_flow2.AuthenticationError) as cm:
+            unified_login_flow2.process(
+                self.api_url,
+                product="my_app",
+                browser_open_callback=lambda url: True,
+            )
```

Comment:
I'm a little suprised bout how cm is not used in the clock here.

Reply:
Yes, I agree, this is a bit confusing. The cm variable is used afterward. Outside this block. It took me some time to accept that too.
https://docs.python.org/3.9/library/unittest.html#unittest.TestCase.assertRaises